CMakeLists.txt

```diff
@@ -37,6 +37,7 @@ option(ENABLE_CCACHE "Whether to use ccache to speed up compiling" ON)
 option(ENABLE_ASAN "Whether to turn AddressSanitizer ON or OFF" OFF)
 option(ENABLE_TESTING "Whether to turn unit test ON or OFF" ON)
 option(ENABLE_UBSAN "Whether to turn Undefined Behavior Sanitizer ON or OFF" OFF)
+option(ENABLE_COVERAGE "Enable coverage reporting" ON)
```

Comment:
Is it necessary to enable this by default?

Reply:
> Is it necessary to enable this by default?

Yes, disable default better.